doc/modules/feature_extraction.rst

```diff
@@ -101,6 +101,81 @@ memory the ``DictVectorizer`` class uses a ``scipy.sparse`` matrix by
 default instead of a ``numpy.ndarray``.
 
 
+.. _column_transformer:
+
+ColumnTransformer for heterogeneous data
+========================================
+
+.. warning::
+
+    The :class:`experimental.ColumnTransformer <sklearn.experimental.ColumnTransformer>`
+    class is experimental and the API is subject to change.
+
+Many datasets contain features of different types, say text, floats, and dates,
+where each type of feature requires separate preprocessing or feature extraction steps.
+Often it is easiest to preprocess data before applying scikit-learn methods,
+for example using `pandas <http://pandas.pydata.org/>`__.
+The :class:`~sklearn.experimental.ColumnTransformer` is a convenient way to perform heterogeneous
+preprocessing on data columns within a scikit-learn pipeline; e.g. when one wants
+to adjust preprocessing parameters within a grid search or to
+avoid leaking statistics from test data to training data.
+The :class:`~sklearn.experimental.ColumnTransformer` works on arrays, sparse matrices,
+and pandas DataFrames.
+
+To each column, a different transformation can be applied, such as
+preprocessing or a specific feature extraction method::
+
+  >>> import pandas as pd
+  >>> X = pd.DataFrame(
+  ...     {'city': ['London', 'London', 'Paris', 'New York'],
+  ...      'title': ["His Last Bow", "How Watson Learned the Trick",
+  ...                "A Moveable Feast", "The Great Gatsby"]})
+
+For this data, we might want to encode ``'city'`` column as a categorical variable,
+but apply a :class:`feature_extraction.text.CountVectorizer <sklearn.feature_extraction.text.CountVectorizer>`
+to the ``'title'`` column.
+As we might use multiple feature extraction methods on the same column, we give each
+transformer a unique name, say ``'city_category'`` and ``'title_bow'``::
+
+  >>> from sklearn.experimental import ColumnTransformer
+  >>> from sklearn.feature_extraction.text import CountVectorizer
+  >>> column_trans = ColumnTransformer(
+  ...     [('city_category', CountVectorizer(analyzer=lambda x: [x]), 'city'),
+  ...      ('title_bow', CountVectorizer(), 'title')])
+
+  >>> column_trans.fit(X) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+  ColumnTransformer(n_jobs=1, transformer_weights=None,
+      transformers=...)
+
+  >>> column_trans.get_feature_names() == [
+  ...    'city_category__London', 'city_category__New York', 'city_category__Paris',
+  ...    'title_bow__bow', 'title_bow__feast', 'title_bow__gatsby',
+  ...    'title_bow__great', 'title_bow__his', 'title_bow__how', 'title_bow__last',
+  ...    'title_bow__learned', 'title_bow__moveable', 'title_bow__the',
+  ...    'title_bow__trick', 'title_bow__watson']
```

Comment:
Wouldn't this be more readable to simply print the result of column_trans, rather than using the "=="? (which I suspect is there to ease doctesting)